pkg/federation/fed_single_cluster_install.go

```diff
@@ -41,58 +35,54 @@ func getenv(key, fallback string) string {
 }
 
 func TestSingleClusterFed(t *testing.T) {
-	defer cleanupSingleClusterFed()
+	NewTest(t).LegacyID("T31", "T8").Groups(ARM, Full).Run(func(t TestHelper) {
```

Comment:
Because tests in `_test.go` files can't be referenced outside their respective packages, there is no way to have a centralized list of tests for each group. That's why I moved the information about the group into the individual tests. 

If we need to see a list of all the tests in a group, we'll need to get that information in a different way. Maybe by parsing the _test files? 

But maybe we don't need a list at all, since every time you run the tests of a certain group, you see which tests belong to it and which don't (currently, tests that don't belong to the specified group are skipped (and this is logged)).

Reply:
Great, I agree with you maybe we don't need the list because we are going to see the test that runs and the test that are skipped

Reply:
Actually, we may need to implement test groups differently for other reasons. I just realized that skipping tests isn't a good approach, because the suite setup/teardown runs regardless if all the tests in the suite are skipped.  Unless we say that all the tests in a suite can only belong to a single group. In that case, we can easily skip the entire suite.

Reply:
We can flush these sorts of minor issues later.

Reply:
Maybe we can use build tags for test groups (this was actually my initial idea).

And we can include a `--dry-run` option that doesn't actually run the tests, but instead just prints out the names of all the tests that would run. This way, we can easily get the exact list of tests in a group.